src/zjs_callbacks.c

```diff
@@ -315,17 +312,21 @@ void zjs_remove_callback(int32_t id)
     }
 }
 
-void zjs_signal_callback(int32_t id)
+void zjs_signal_callback(int32_t id, void* args, uint32_t size)
 {
-    if (id != -1 && cb_map[id]) {
-#ifdef DEBUG_BUILD
-        if (cb_map[id]->type == CALLBACK_TYPE_JS) {
-            DBG_PRINT("signaling JS callback id %ld\n", id);
-        } else {
-            DBG_PRINT("signaling C callback id %ld\n", id);
+    if (args && size) {
+        // signal with args
+        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
```

Comment:
Still a good idea just to be safe. Then you don't *need* to document that they have to use something divisible by four; it will work even if they don't.